src/gc/boehm.cr

```diff
@@ -93,7 +97,7 @@ lib LibGC
 
   fun push_all_eager = GC_push_all_eager(bottom : Void*, top : Void*)
 
-  {% if flag?(:preview_mt) || flag?(:win32) %}
+  {% if flag?(:preview_mt) || flag?(:win32) || compare_versions(VERSION, "8.3.0") >= 0 %}
```

Comment:
**thought:** Since we're using version detection, we could map binding availability to symbol availability in the library.

```suggestion
  {% if flag?(:preview_mt) || flag?(:win32) || compare_versions(VERSION, "8.2.0") >= 0 %}
```

The other conditions are correct. With `preview_mt` and on Windows we always need those bindings.

Otherwise we still wouldn't *use* the bindings for versions below 8.3.0, but the bindings should be there. Of course this adds complexity to the calling condition because we can't simply check whether the binding is defined. Instead of `LibGC.has_method?(:get_my_stackbottom)` we would need to use `flag?(:preview_mt) || flag?(:win32) || compare_versions(VERSION, "8.2.0") >= 0`.

So this is technically more accurate, but increases code complexity. It's a trade off, and I'm fine with either solution.

Reply:
I was on the impression that the `fiber_qsort` benchmark that was crashing (see link above) was noticeably slower with the functions, but after testing again, the speed is roughly the same :shrug:

Let's enable them based if they're available (>= 8.2.0).